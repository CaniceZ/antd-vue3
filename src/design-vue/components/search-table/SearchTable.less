.ygp-search-table {
  &-form {
    padding: 16px;
    padding-bottom: 0;
    background: #f7f8fa;
    margin-bottom: 8px;

    .ygp-form {
      .ygp-form-item {
        margin-bottom: 18px;
        &.ant-form-item-with-help {
          margin-bottom: 0;
        }
      }
    }
  }

  &-fit {
    min-height: 100%;
    height: 100%;
    min-width: 0;
    display: flex;
    flex-direction: column;

    & > .ygp-search-table-content {
      min-height: 0;
      height: 100%;

      & > .ant-spin-nested-loading {
        flex: 1;
        min-height: 0;
        height: 100%;
        & > .ant-spin-container {
          height: 100%;
          display: flex;
          flex-direction: column;
          & > .ygp-form {
            height: 100%;
            flex: 1;
            min-height: 0;
            & > .ygp-table {
              height: 100%;
              & > .ant-spin-nested-loading {
                height: 100%;
                & > .ant-spin-container {
                  height: 100%;
                  display: flex;
                  flex-direction: column;
                  & > .ant-table {
                    height: 100%;
                    min-height: 0;
                    & > .ant-table-container {
                      height: 100%;
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}
