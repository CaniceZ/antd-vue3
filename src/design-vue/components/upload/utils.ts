export const extname = (url = '') => {
  const temp = url.split('/')
  const filename = temp[temp.length - 1]
  const filenameWithoutSuffix = filename.split(/#|\?/)[0]
  return (/\.[^./\\]*$/.exec(filenameWithoutSuffix) || [''])[0]
}

const isImageFileType = type => !!type && type.indexOf('image/') === 0

export const isImageUrl = file => {
  if (isImageFileType(file.type)) {
    return true
  }
  const url = file.thumbUrl || file.url
  const extension = extname(url)
  if (
    /^data:image\//.test(url) ||
    /(webp|svg|png|gif|jpg|jpeg|jfif|bmp|dpg|ico)$/i.test(extension)
  ) {
    return true
  }
  if (/^data:/.test(url)) {
    // other file types of base64
    return false
  }
  if (extension) {
    // other file types which have extension
    return false
  }
  return true
}

export const getFileType = file => {
  if (!file) {
    return
  }
  const attachmentUrl = file.url

  const type = extname(attachmentUrl)
  let fileType = ''
  if (
    isImageFileType(file.type) ||
    /(webp|svg|png|gif|jpg|jpeg|jfif|bmp|dpg|ico)$/i.test(type)
  ) {
    fileType = 'image'
  } else if (type === '.pdf') {
    fileType = 'pdf'
  } else if (['.doc', '.dot'].includes(type)) {
    fileType = 'word'
  } else if (['.xls', '.xlsx'].includes(type)) {
    fileType = 'excel'
  } else {
    fileType = 'file'
  }
  return fileType
}

export const imageDimensions = file =>
  new Promise((resolve, reject) => {
    const img = new Image()

    img.onload = () => {
      const { naturalWidth: width, naturalHeight: height } = img
      resolve({ width, height })
    }

    img.onerror = () => {
      reject('There was some problem with the image.')
    }

    img.src = URL.createObjectURL(file)
  })

export const isUrl = (url: string) => {
  const regex = /^(http|https)/
  return regex.test(url.toLowerCase())
}
