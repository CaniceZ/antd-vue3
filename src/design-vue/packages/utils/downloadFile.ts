/**
 * 下载文件
 * @param url 下载路径
 * @returns void
 */
export default function downloadFile(url: string) {
  if (!url) return
  const a = document.createElement('a')
  a.href = url
  a.click()
}
