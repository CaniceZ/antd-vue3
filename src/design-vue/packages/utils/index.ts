import omit from './omit'
import pick from './pick'
import downloadFile from './downloadFile'
import { isCiphertext, getCiphertext, getCiphertextField } from './ciphertext'
import { needCLodop, loadCLodop, getLodop } from './LodopFuncs'
import { importFile, importResultPolling } from './excel'
import filterUnit from './filterUnit'
import filterMoney from './filterMoney'

export {
  omit,
  pick,
  downloadFile,
  isCiphertext,
  getCiphertext,
  getCiphertextField,
  needCLodop,
  loadCLodop,
  getLodop,
  importFile,
  importResultPolling,
  filterUnit,
  filterMoney
}
