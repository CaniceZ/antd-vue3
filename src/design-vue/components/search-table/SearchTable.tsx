import { computed, defineComponent, ref, VNode, watch, camelize } from 'vue'
import { UpOutlined, DownOutlined } from '@ant-design/icons-vue'
import { useGlobalConfig } from '../../plugins/global-config'
import { filterSlots, getPrefix, parseAttr } from '../../_utils/common'
import { omit } from '../../packages//utils'
import { GlobalComponentConstructor, PropsType } from '../../types/ts-helpers'
import { TableExpose, TableProps, TableSlots } from '../table'
import { FormExpose, FormProps, FormSlots } from '../form'
import kebabCase from 'lodash-es/kebabCase'

const [prefixName, prefixCls] = getPrefix('search-table')

const SearchTable = defineComponent({
  name: prefixName,
  inheritAttrs: false,
  props: {
    visibleLine: {
      type: Number,
      default: 2
    },
    wrapperClassName: [String, Object, Array],
    wrapperStyle: [String, Object]
  },
  emits: ['search'],
  setup(props, { slots, emit, expose, attrs }) {
    const globalConfig = useGlobalConfig()
    const formRef = ref<FormExpose>()
    const tableRef = ref<TableExpose>()

    const model = computed(() => attrs['model'] || {})

    const visible = ref(false)
    const newProps = computed(() => {
      const prefix = `form-`
      return Object.keys(attrs).reduce(
        ([formProps, tableProps], key) => {
          const reg = new RegExp(`^${prefix}(.*?)$`)
          const [, matchName] = kebabCase(key).match(reg) || []
          if (matchName || ['model', 'items'].includes(key)) {
            return [
              {
                ...formProps,
                [matchName ? camelize(matchName) : key]: attrs[key]
              },
              tableProps
            ]
          } else {
            return [formProps, { ...tableProps, [key]: attrs[key] }]
          }
        },
        [{}, {}]
      )
    })
    const formProps = computed(() => newProps.value[0])
    const tableProps = computed(() => newProps.value[1])
    const hideNames = computed(() => {
      const hideNames: string[] = []
      let totalSpan = 0
      formProps.value['items']?.forEach(item => {
        let span = Math.min(24, formRef.value?.getItemSpan(item.span) || 0)
        if (item.row) {
          span = 24 + (24 - (totalSpan % 24))
        } else if (item.newline) {
          span = 24 - ((totalSpan + span) % 24) + span
        }
        if ((totalSpan += span) > 24 * props.visibleLine) {
          hideNames.push(item.name)
        }
      })
      return hideNames
    })

    const stickyAttrs = computed(() => {
      const sticky = globalConfig.appConfig?.searchTableSticky
      if (sticky) {
        return { sticky }
      }
      return {}
    })

    function toggle() {
      visible.value = !visible.value
    }

    function expand() {
      visible.value = true
    }

    function collapse() {
      visible.value = false
    }

    function setTableHeader() {
      tableRef.value?.setHeader()
    }
    function setTableProps(option) {
      tableRef.value?.setTableProps(option)
    }

    async function onSearch() {
      await formRef.value?.validate()
      emit('search')
    }

    function onExport() {
      const exportFn = attrs['onExport']
      if (typeof exportFn === 'function') {
        exportFn()
      }
    }

    async function onReset() {
      await formRef.value?.reset()
      let res
      const resetFn = attrs['onReset']
      if (typeof resetFn === 'function') {
        res = await resetFn()
      }
      if (res !== false) {
        await onSearch()
      }
    }

    expose({
      search: onSearch,
      export: onExport,
      reset: onReset,
      toggle,
      expand,
      collapse,
      form: formRef,
      table: tableRef,
      setTableHeader: setTableHeader,
      setTableProps: setTableProps
    })

    watch([visible, hideNames, formRef], ([visible, names, form]) => {
      if (form) {
        form.showAll()
        visible ? form.show(names) : form.hide(names)
      }
    })

    return () => {
      const formSlots = filterSlots('form-', slots)
      const tableSlots = omit(
        slots,
        Object.keys(formSlots)
          .map(key => `form-${key}`)
          .concat(['top', 'top-action', 'tabs'])
      )
      return (
        <div
          style={props.wrapperStyle}
          class={[
            prefixCls,
            props.wrapperClassName,
            { [`${prefixCls}-fit`]: parseAttr(attrs.fit) }
          ]}
        >
          {(slots.top || slots['top-action']) && (
            <div style="margin-bottom: 16px;">
              {slots.top?.() ||
                (slots['top-action'] && (
                  <y-space
                    v-slots={{
                      default: slots['top-action']
                    }}
                  />
                ))}
            </div>
          )}
          <div class={`${prefixCls}-form`}>
            {slots.form?.() || (
              <y-form
                ref={formRef}
                defaultSpan={6}
                {...formProps.value}
                model={model.value}
                v-slots={{
                  action: () => (
                    <>
                      <y-btn
                        label="查询"
                        onClick={onSearch}
                        primary
                        loading={attrs['loading']}
                      />
                      {attrs['onExport'] && (
                        <y-btn
                          label="导出"
                          onClick={onExport}
                          {...((attrs['export-btn-props'] as object) ||
                            (attrs['exportBtnProps'] as object) ||
                            {})}
                        />
                      )}
                      {(formProps.value?.['items'] || []).length > 0 && (
                        <y-btn
                          link
                          label="重置筛选条件"
                          onClick={onReset}
                          disabled={attrs['loading']}
                        />
                      )}
                      {hideNames.value.length > 0 && (
                        <y-btn onClick={toggle} link>
                          {visible.value ? '收起' : '展开'}
                          {visible.value ? <UpOutlined /> : <DownOutlined />}
                        </y-btn>
                      )}
                    </>
                  ),
                  ...formSlots
                }}
              />
            )}
          </div>
          {slots.tabs && (
            <div class={`${prefixCls}-tabs`}>{slots.tabs?.()}</div>
          )}
          <div class={`${prefixCls}-content`}>
            {slots.default?.({
              tableProps: { ...stickyAttrs.value, ...tableProps.value },
              tableRef,
              tableSlots
            }) || (
              <y-table
                {...stickyAttrs.value}
                ref={tableRef}
                {...tableProps.value}
                v-slots={tableSlots}
              />
            )}
          </div>
        </div>
      )
    }
  }
})

type PrefixForm<T> = {
  [K in keyof T as K extends string ? `form-${K}` : never]: T[K]
}
export type SearchTableProps = PropsType<typeof SearchTable> &
  PropsType<TableProps> &
  Pick<FormProps, 'model' | 'items'> &
  PrefixForm<FormProps>
export type SearchTableExpose = {
  search: () => Promise<any>
  export(): void
  reset: () => Promise<any>
  toggle(): void
  expand(): void
  collapse(): void
  form: FormExpose
  table: TableExpose
  setTableHeader(): void
}
export type SearchTableSlots = TableSlots &
  Readonly<
    {
      top?(): VNode[]
      'top-action'?(): VNode[]
      tabs?(): VNode[]
      [key: `form-${string}`]: () => VNode[]
    } & PrefixForm<FormSlots>
  >
export default SearchTable as GlobalComponentConstructor<
  SearchTableProps,
  SearchTableSlots
>
