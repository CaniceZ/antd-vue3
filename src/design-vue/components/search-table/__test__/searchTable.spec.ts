import { mount } from '@vue/test-utils'
import { asyncExpect } from '../../../tests/utils'
import SearchTable from '..'
import Ygpd from '../../../index'
import {
  SEARCH_TABLE_TEST_DATA,
  SEARCH_TABLE_COLUMNS,
  SEARCH_TABLE_ITEMS
} from './searchTable'

describe('SearchTable', () => {
  const { getComputedStyle } = window
  window.getComputedStyle = elt => getComputedStyle(elt)

  // 关于插槽能够成功渲染的用例
  it('SearchTable component table and form component slot render is ok', async () => {
    const wrapper = mount(SearchTable as any, {
      props: {
        ref: 'crud',
        rowKey: 'id',
        data: SEARCH_TABLE_TEST_DATA,
        items: [
          {
            name: 'realname',
            label: '姓名',
            newline: true,
            placeholder: '设置`newline: true`换行'
          }
        ],
        columns: SEARCH_TABLE_COLUMNS
      },
      global: {
        plugins: [Ygpd]
        // renderStubDefaultSlot: true,
      },
      slots: {
        default: '<div class="slot-table">这是自定义table的插槽</div>',
        form: '<div class="slot-form">这是自定义form的插槽</div>',
        tabs: '<div class="slot-tabs">这是自定义tabs的插槽</div>'
      }
      // shallow: true,
    })

    // 断言 获取渲染在对应form插槽的元素，如果获取成功则为true 通过
    expect(wrapper.find('.ygp-search-table-form .slot-form').exists()).toBe(
      true
    )

    // 断言 获取渲染在对应table插槽的元素，如果获取成功则为true 通过
    expect(wrapper.find('.ygp-search-table-content .slot-table').exists()).toBe(
      true
    )

    // 断言 获取渲染在对应tabs插槽的元素，如果获取成功则为true 通过
    expect(wrapper.find('.ygp-search-table-tabs .slot-tabs').exists()).toBe(
      true
    )

    await asyncExpect(() => {
      expect(wrapper.html()).toMatchSnapshot()
    }, 0)
  })

  // 使用y-form能否成功渲染的用例
  it('SearchTable component YForm component render is ok', async () => {
    const wrapper = mount(SearchTable as any, {
      props: {
        ref: 'crud',
        rowKey: 'id',
        data: SEARCH_TABLE_TEST_DATA,
        items: SEARCH_TABLE_ITEMS,
        columns: SEARCH_TABLE_COLUMNS,
        'form-name': 'basic'
      },
      global: {
        plugins: [Ygpd]
        // renderStubDefaultSlot: true,
      },
      slots: {
        default: '<div class="slot-table">这是自定义table的插槽</div>',
        tabs: '<div class="slot-tabs">这是自定义tabs的插槽</div>'
      }
      // shallow: true,
    })

    await asyncExpect(async () => {
      const formItem = await wrapper.findAll('.ygp-form-item')
      // 断言 form表单一开始生成元素，在超过一定数量的时候会将一部分元素先隐藏起来，如果获取的formitem - 1 数量大于了0小于需要生成的最大item数，则为真
      // 也证明了收起的功能是有用的
      expect(
        formItem.length - 1 > 0 &&
          formItem.length - 1 < SEARCH_TABLE_ITEMS.length
      ).toBe(true)

      // 点击展开生成对应的元素
      await wrapper.findAll('.action_box .ygp-btn')[2].trigger('click')

      // 断言 展开后生成的formitem 跟我们引入的items，数量一直，则为真
      // 也证明了展开是可行的
      expect(
        wrapper.findAll('.ygp-form-item').length - 1 ===
          SEARCH_TABLE_ITEMS.length
      ).toBe(true)

      expect(wrapper.html()).toMatchSnapshot()
    }, 0)
  })

  // 使用y-table能否成功渲染的用例
  it('SearchTable component YTable component render is ok', async () => {
    const wrapper = mount(SearchTable as any, {
      props: {
        ref: 'crud',
        rowKey: 'id',
        data: SEARCH_TABLE_TEST_DATA,
        columns: SEARCH_TABLE_COLUMNS
      },
      global: {
        plugins: [Ygpd]
        // renderStubDefaultSlot: true,
      },
      slots: {
        form: '<div class="slot-table">这是自定义table的插槽</div>',
        tabs: '<div class="slot-tabs">这是自定义tabs的插槽</div>'
      },
      shallow: true
    })

    const tableEle = wrapper.find('y-table-stub')
    // 断言 没有default 插槽，希望能生成y-table组件，如果为真则是渲染了y-table组件
    expect(wrapper.findComponent('y-table-stub').exists()).toBe(true)

    // 断言 根据传入的data数据如果为两个，则为[object Object],[object Object]，如果相等则证明打data传入成功
    expect(tableEle.attributes('data')).toBe('[object Object],[object Object]')

    // 断言 根据传入的columns数据如果为两个，则为4个[object Object]，如果相等则证明打columns传入成功
    expect(tableEle.attributes('columns')).toBe(
      '[object Object],[object Object],[object Object],[object Object]'
    )

    await wrapper.setProps({
      data: [
        {
          id: '1',
          name: 'iPhone13',
          status: '已发布'
        }
      ],
      columns: [
        {
          name: 'realname',
          label: '姓名',
          newline: true,
          placeholder: '设置`newline: true`换行'
        }
      ]
    })

    // 断言 根据传入的data数据如果为1个，则为[object Object]，如果相等则证明打data传入成功
    expect(tableEle.attributes('data')).toBe('[object Object]')

    // 断言 根据传入的columns数据如果为1个，则为[object Object]，如果相等则证明打columns传入成功
    expect(tableEle.attributes('columns')).toBe('[object Object]')

    // expect(tableEle.attributes('rowKey')).toBe('id')
    // console.log(tableEle.attributes('rowKey'))

    // console.log(wrapper.find('y-table-stub').html())

    await asyncExpect(async () => {
      expect(wrapper.html()).toMatchSnapshot()
    }, 0)
  })
})
