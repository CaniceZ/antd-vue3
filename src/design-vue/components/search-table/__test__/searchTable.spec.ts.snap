// Vitest Snapshot v1

exports[`SearchTable > SearchTable component YForm component render is ok 1`] = `
"<div class=\\"ygp-search-table\\">
  <!---->
  <div class=\\"ygp-search-table-form\\">
    <form class=\\"ant-form ant-form-horizontal ygp-form\\">
      <div class=\\"ant-row ygp-row\\" style=\\"margin-left: -8px; margin-right: -8px;\\">
        <!---->
        <div class=\\"ant-col ant-col-6 ygp-col\\" style=\\"padding-left: 8px; padding-right: 8px;\\">
          <div class=\\"ant-row ant-form-item ygp-form-item\\" trigger=\\"change\\">
            <div class=\\"ant-col ant-form-item-label\\" style=\\"width: 120px;\\"><label for=\\"basic_realname\\" class=\\"\\" title=\\"\\"><span>姓名<!----></span>
                <!---->
              </label></div>
            <div class=\\"ant-col ant-form-item-control\\">
              <div class=\\"ant-form-item-control-input\\">
                <div class=\\"ant-form-item-control-input-content\\"><span class=\\"ant-input-affix-wrapper ygp-input\\"><!----><input placeholder=\\"设置\`newline: true\`换行\\" type=\\"text\\" id=\\"basic_realname\\" options=\\"\\" listtype=\\"link\\" class=\\"ant-input\\"><span class=\\"ant-input-suffix\\"><span role=\\"button\\" aria-label=\\"close-circle\\" tabindex=\\"-1\\" class=\\"anticon anticon-close-circle ant-input-clear-icon-hidden ant-input-clear-icon-has-suffix ant-input-clear-icon\\"><svg focusable=\\"false\\" class=\\"\\" data-icon=\\"close-circle\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" viewBox=\\"64 64 896 896\\"><path d=\\"M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z\\"></path></svg></span>
                  <!----></span></span>
                </div>
                <!---->
              </div>
              <!---->
              <!---->
            </div>
          </div>
        </div>
        <div class=\\"ant-col ant-col-24\\" style=\\"padding-left: 8px; padding-right: 8px;\\">
          <!---->
        </div>
        <!---->
        <div class=\\"ant-col ant-col-24 ygp-col\\" style=\\"padding-left: 8px; padding-right: 8px;\\">
          <div class=\\"ant-row ant-form-item ygp-form-item\\" trigger=\\"change\\">
            <div class=\\"ant-col ant-form-item-label\\" style=\\"width: 120px;\\"><label for=\\"basic_password\\" class=\\"\\" title=\\"\\"><span>密码<!----></span>
                <!---->
              </label></div>
            <div class=\\"ant-col ant-form-item-control\\">
              <div class=\\"ant-form-item-control-input\\">
                <div class=\\"ant-form-item-control-input-content\\"><span class=\\"ant-input-affix-wrapper ygp-input\\"><!----><input placeholder=\\"请输入\\" type=\\"text\\" id=\\"basic_password\\" options=\\"\\" listtype=\\"link\\" class=\\"ant-input\\"><span class=\\"ant-input-suffix\\"><span role=\\"button\\" aria-label=\\"close-circle\\" tabindex=\\"-1\\" class=\\"anticon anticon-close-circle ant-input-clear-icon-hidden ant-input-clear-icon-has-suffix ant-input-clear-icon\\"><svg focusable=\\"false\\" class=\\"\\" data-icon=\\"close-circle\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" viewBox=\\"64 64 896 896\\"><path d=\\"M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z\\"></path></svg></span>
                  <!----></span></span>
                </div>
                <!---->
              </div>
              <!---->
              <!---->
            </div>
          </div>
        </div>
        <div class=\\"ant-col ant-col-24\\" style=\\"padding-left: 8px; padding-right: 8px;\\">
          <!---->
        </div>
        <!---->
        <div class=\\"ant-col ant-col-24 ygp-col\\" style=\\"padding-left: 8px; padding-right: 8px;\\">
          <div class=\\"ant-row ant-form-item ygp-form-item\\" trigger=\\"change\\">
            <div class=\\"ant-col ant-form-item-label\\" style=\\"width: 120px;\\"><label for=\\"basic_confirmPassword\\" class=\\"\\" title=\\"\\"><span>确认密码<!----></span>
                <!---->
              </label></div>
            <div class=\\"ant-col ant-form-item-control\\">
              <div class=\\"ant-form-item-control-input\\">
                <div class=\\"ant-form-item-control-input-content\\"><span class=\\"ant-input-affix-wrapper ygp-input\\"><!----><input placeholder=\\"请输入\\" type=\\"text\\" id=\\"basic_confirmPassword\\" options=\\"\\" listtype=\\"link\\" class=\\"ant-input\\"><span class=\\"ant-input-suffix\\"><span role=\\"button\\" aria-label=\\"close-circle\\" tabindex=\\"-1\\" class=\\"anticon anticon-close-circle ant-input-clear-icon-hidden ant-input-clear-icon-has-suffix ant-input-clear-icon\\"><svg focusable=\\"false\\" class=\\"\\" data-icon=\\"close-circle\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" viewBox=\\"64 64 896 896\\"><path d=\\"M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z\\"></path></svg></span>
                  <!----></span></span>
                </div>
                <!---->
              </div>
              <!---->
              <!---->
            </div>
          </div>
        </div>
        <div class=\\"ant-col ant-col-24\\" style=\\"padding-left: 8px; padding-right: 8px;\\">
          <!---->
        </div>
        <!---->
        <div class=\\"ant-col ant-col-24 ygp-col\\" style=\\"padding-left: 8px; padding-right: 8px;\\">
          <div class=\\"ant-row ant-form-item ygp-form-item\\" trigger=\\"change\\">
            <div class=\\"ant-col ant-form-item-label\\" style=\\"width: 120px;\\"><label for=\\"basic_password1\\" class=\\"\\" title=\\"\\"><span>密码1<!----></span>
                <!---->
              </label></div>
            <div class=\\"ant-col ant-form-item-control\\">
              <div class=\\"ant-form-item-control-input\\">
                <div class=\\"ant-form-item-control-input-content\\"><span class=\\"ant-input-affix-wrapper ygp-input\\"><!----><input placeholder=\\"请输入\\" type=\\"text\\" id=\\"basic_password1\\" options=\\"\\" listtype=\\"link\\" class=\\"ant-input\\"><span class=\\"ant-input-suffix\\"><span role=\\"button\\" aria-label=\\"close-circle\\" tabindex=\\"-1\\" class=\\"anticon anticon-close-circle ant-input-clear-icon-hidden ant-input-clear-icon-has-suffix ant-input-clear-icon\\"><svg focusable=\\"false\\" class=\\"\\" data-icon=\\"close-circle\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" viewBox=\\"64 64 896 896\\"><path d=\\"M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z\\"></path></svg></span>
                  <!----></span></span>
                </div>
                <!---->
              </div>
              <!---->
              <!---->
            </div>
          </div>
        </div>
        <div class=\\"ant-col ant-col-24\\" style=\\"padding-left: 8px; padding-right: 8px;\\">
          <!---->
        </div>
        <!---->
        <div class=\\"ant-col ant-col-24 ygp-col\\" style=\\"padding-left: 8px; padding-right: 8px;\\">
          <div class=\\"ant-row ant-form-item ygp-form-item\\" trigger=\\"change\\">
            <div class=\\"ant-col ant-form-item-label\\" style=\\"width: 120px;\\"><label for=\\"basic_confirmPassword1\\" class=\\"\\" title=\\"\\"><span>确认密码1<!----></span>
                <!---->
              </label></div>
            <div class=\\"ant-col ant-form-item-control\\">
              <div class=\\"ant-form-item-control-input\\">
                <div class=\\"ant-form-item-control-input-content\\"><span class=\\"ant-input-affix-wrapper ygp-input\\"><!----><input placeholder=\\"请输入\\" type=\\"text\\" id=\\"basic_confirmPassword1\\" options=\\"\\" listtype=\\"link\\" class=\\"ant-input\\"><span class=\\"ant-input-suffix\\"><span role=\\"button\\" aria-label=\\"close-circle\\" tabindex=\\"-1\\" class=\\"anticon anticon-close-circle ant-input-clear-icon-hidden ant-input-clear-icon-has-suffix ant-input-clear-icon\\"><svg focusable=\\"false\\" class=\\"\\" data-icon=\\"close-circle\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" viewBox=\\"64 64 896 896\\"><path d=\\"M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z\\"></path></svg></span>
                  <!----></span></span>
                </div>
                <!---->
              </div>
              <!---->
              <!---->
            </div>
          </div>
        </div>
        <div class=\\"ant-col ant-col-24\\" style=\\"padding-left: 8px; padding-right: 8px;\\">
          <!---->
        </div>
      </div>
      <div class=\\"ant-row ygp-row\\">
        <div class=\\"action_box\\">
          <div class=\\"ant-row ant-form-item ygp-form-item\\" trigger=\\"change\\">
            <div class=\\"ant-col ant-form-item-label\\" style=\\"width: 120px;\\"><label class=\\"ant-form-item-no-colon\\" title=\\"\\"><span> <!----></span>
                <!---->
              </label></div>
            <div class=\\"ant-col ant-form-item-control\\">
              <div class=\\"ant-form-item-control-input\\">
                <div class=\\"ant-form-item-control-input-content\\">
                  <div>
                    <div class=\\"ant-space ant-space-horizontal ant-space-align-center ygp-space\\">
                      <div class=\\"ant-space-item\\" style=\\"margin-right: 8px;\\"><button class=\\"ant-btn ant-btn-primary ygp-btn\\" type=\\"button\\">
                          <transition-stub name=\\"ant-btn-loading-icon-motion\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                            <!---->
                          </transition-stub><span>查 询</span>
                        </button></div>
                      <!---->
                      <div class=\\"ant-space-item\\" style=\\"margin-right: 8px;\\">
                        <!---->
                      </div>
                      <!---->
                      <div class=\\"ant-space-item\\" style=\\"margin-right: 8px;\\"><button class=\\"ant-btn ant-btn-link ygp-btn\\" type=\\"button\\">
                          <transition-stub name=\\"ant-btn-loading-icon-motion\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                            <!---->
                          </transition-stub><span>重置筛选条件</span>
                        </button></div>
                      <!---->
                      <div class=\\"ant-space-item\\"><button class=\\"ant-btn ant-btn-link ygp-btn\\" type=\\"button\\">
                          <transition-stub name=\\"ant-btn-loading-icon-motion\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                            <!---->
                          </transition-stub><span>收起</span><span role=\\"img\\" aria-label=\\"up\\" class=\\"anticon anticon-up\\"><svg focusable=\\"false\\" class=\\"\\" data-icon=\\"up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" viewBox=\\"64 64 896 896\\"><path d=\\"M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z\\"></path></svg></span>
                        </button></div>
                      <!---->
                    </div>
                  </div>
                </div>
                <!---->
              </div>
              <!---->
              <!---->
            </div>
          </div>
          <!---->
        </div>
      </div>
    </form>
  </div>
  <div class=\\"ygp-search-table-tabs\\">
    <div class=\\"slot-tabs\\">这是自定义tabs的插槽</div>
  </div>
  <div class=\\"ygp-search-table-content\\">
    <div class=\\"slot-table\\">这是自定义table的插槽</div>
  </div>
</div>"
`;

exports[`SearchTable > SearchTable component YTable component render is ok 1`] = `
"<div class=\\"ygp-search-table\\">
  <!---->
  <div class=\\"ygp-search-table-form\\">
    <div class=\\"slot-table\\">这是自定义table的插槽</div>
  </div>
  <div class=\\"ygp-search-table-tabs\\">
    <div class=\\"slot-tabs\\">这是自定义tabs的插槽</div>
  </div>
  <div class=\\"ygp-search-table-content\\">
    <y-table-stub data=\\"[object Object]\\" middle=\\"false\\" large=\\"false\\" propmap=\\"[object Object]\\" ellipsis=\\"false\\" bordered=\\"false\\" nopagination=\\"false\\" editable=\\"false\\" configthead=\\"false\\" seq=\\"false\\" defaultexpandallrows2=\\"false\\" nohover=\\"false\\" noresize=\\"false\\" fit=\\"false\\" rowkey=\\"id\\" columns=\\"[object Object]\\"></y-table-stub>
  </div>
</div>"
`;

exports[`SearchTable > SearchTable component table and form component slot render is ok 1`] = `
"<div class=\\"ygp-search-table\\">
  <!---->
  <div class=\\"ygp-search-table-form\\">
    <div class=\\"slot-form\\">这是自定义form的插槽</div>
  </div>
  <div class=\\"ygp-search-table-tabs\\">
    <div class=\\"slot-tabs\\">这是自定义tabs的插槽</div>
  </div>
  <div class=\\"ygp-search-table-content\\">
    <div class=\\"slot-table\\">这是自定义table的插槽</div>
  </div>
</div>"
`;
